### PR TITLE
[ETH-245] ERC20JoinPolicy & DelegatedAccessRegistry

### PR DESCRIPTION
**Pull request by bearni95**
This PR introduces two new smart contracts to enable users to join a stream based on the amount of DATA token that they hold.

The smart contracts can be found in [their corresponding branch](https://github.com/streamr-dev/network-contracts/pull/202)

*DelegatedAccessRegistry*
Very simple smart contract deployed at `0x52278782360728dC8516253fF0415B5c66f2abd7` that maps the Metamask address of each user to their session addresses. It is needed to verify that the session wallet requesting permissions is associated with the Metamask wallet holding those funds.

In order to authorize the session wallet on the smart contract the user must select any room and run the command `/authorize`. A Metamask dialog will prompt for the contract call, associating the session with the metamask addresses

*ERC20JoinPolicy*
Right now there is one deployed, functional ERC20JoinPolicy smart contract at `0x1A88D7895A32b7fC4bFB169987F892a173e889cf`. It is authorized to grant `SUBSCRIBE` and `PUBLISH` permissions on the stream `0x2f1418bbe3512156175efa3ef466f40df0161990/streamr-chat/room/erc20-join-policy` to any wallet holding more than 1 wei in DATA token in the Polygon network, or to any delegated wallets authorized in the DelegatedAccessRegistry at `0x52278782360728dC8516253fF0415B5c66f2abd7`

Users can request to join an ERC20JoinPolicy governed stream by providing the address of the ERC20JoinPolicy smart contract via the `/join` chat command, after clicking on any room:
```
/join 0x1A88D7895A32b7fC4bFB169987F892a173e889cf
```

### CODE REVIEW

File: src/components/SessionHandler.ts [hunk 1]
```diff
@@ -14,7 +14,6 @@ export default function SessionHandler() {
     const dispatch = useDispatch()
 
     const ensureCorrectNetwork = useEnsureCorrectNetwork()
-
```
Comment (mondoreale) [inline src/components/SessionHandler.ts:17]: I don't like this change. xD

File: src/hooks/useEnsureCorrectNetwork.ts [hunk 2]
```diff
@@ -62,6 +54,7 @@ export const requestNetworkChange = async (
                     alert('Successfully connected to Polygon Network')
                     return requestAdd
                 } catch (err: any) {
+                    console.warn('02', err)
```
Comment (mondoreale) [inline src/hooks/useEnsureCorrectNetwork.ts:57]: Let's not push debug logs out. We may want to write tests for it sometime in the future. Atm it's tech debt, let's say.

File: src/components/pages/Chat/index.tsx [hunk 3]
```diff
@@ -52,6 +54,16 @@ const UnstyledChat = ({ className }: Props) => {
...
 
+    const erc20RequestDelegatedJoin = useEJPRequestDelegatedJoin()
+    const [isMount, setIsMount] = useState(false)
+    useEffect(() => {
+        if (isMount) {
+            return
+        }
+        setIsMount(true)
+        erc20RequestDelegatedJoin({})
+    })
```
Comment (mondoreale) [inline src/components/pages/Chat/index.tsx:65]: No deps for `useEffect`? i.e. `[isMount, erc20RequestDelegatedJoin]`?

If you want to do something on mount only then

```jsx
const erc20RequestDelegatedJoinRef = useRef(erc20RequestDelegatedJoin)

useEffect(() => {
    erc20RequestDelegatedJoinRef.current({})
}, [])
```

File: src/getters/getEnvironmentConfig.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+type EnvironmentConfig = {
+    PolygonNetworkInfo: any
+    StreamrClientConfig: any
+    DelegatedAccessRegistryAddress: string
+}
+
+const ProductionConfig: EnvironmentConfig = {
+    PolygonNetworkInfo: {
+        chainId: 137,
+        chainIdHex: `0x${(137).toString(16)}`,
```
Comment (mondoreale) [inline src/getters/getEnvironmentConfig.ts:12]: `0x89`

File: src/getters/getEnvironmentConfig.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    StreamrClientConfig: {},
+    DelegatedAccessRegistryAddress:
+        '0x52278782360728dC8516253fF0415B5c66f2abd7',
+}
+
+const DevConfig: EnvironmentConfig = {
+    PolygonNetworkInfo: {
+        chainId: 8995,
+        chainIdHex: `0x${(8995).toString(16)}`,
```
Comment (mondoreale) [inline src/getters/getEnvironmentConfig.ts:28]: `0x2323`

File: src/hooks/erc20JoinPolicy/useEJPRequestDelegatedJoin.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+type Inviter = ({ invitee, stream }: Options) => Promise<void>
+
+export default function useEJPRequestDelegatedJoin(): Inviter {
+    const {
+        ethereumProvider,
+        session: { wallet },
+    } = useStore()
+    const ERC20Address = ''
```
Comment (mondoreale) [inline src/hooks/erc20JoinPolicy/useEJPRequestDelegatedJoin.ts:25]: Unused?

**Review by mondoreale — COMMENTED** [review]
Have a look at my notes and then let's have another review round.

**Comment by its-edmund** [comment]
I switched the default branch, make sure to change development to master before merging.

**Pull request closed without merging**